OONIProbeUnitTests/APITest.m

```diff
@@ -68,4 +69,29 @@ -(void)testDownloadUrls{
     }];
 }
 
+- (void)testDeleteJsons {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"testDeleteJsons"];
+    Measurement *existing_1 = [self addMeasurement:EXISTING_REPORT_ID];
+    Measurement *nonexisting_1 = [self addMeasurement:NONEXISTING_REPORT_ID];
+    Measurement *existing_2 = [self addMeasurement:EXISTING_REPORT_ID_2];
+    Measurement *nonexisting_2 = [self addMeasurement:NONEXISTING_REPORT_ID];
+    [self deleteUploadedJsonsWithMeasurementRemover:^(Measurement *measurement) {
+        if ([measurement.report_id isEqualToString:EXISTING_REPORT_ID] || [measurement.report_id isEqualToString:EXISTING_REPORT_ID_2])
+            XCTAssert(true);
+        else
+            XCTAssert(false);
+        //TODO how do i check that this callback has been called for both measurement in the if and fulfill only after that?
```

Comment:
@bassosimone

Reply:
- You can't edit `failures` and `successess` inside that without passing the vars somehow.
- I don't see any fullfill in this code, where would you put them?
- What the failure value stands for?

Reply:
How about then using NSInteger rather than just int? Since a NSInteger instance is a pointer to NSInteger, you are able to change its value, right?

I think I forgot to add a fulfill part. In there I think you should assert that you did see two “success” and zero “failure”. Does this make sense?

(It’s still a rather bad test because it runs for ten seconds while you have an answer much early than that...)

Reply:
https://stackoverflow.com/questions/7962721/assign-a-variable-inside-a-block-to-a-variable-outside-a-block

Reply:
I guess there is no need for the failure var. When it enters the failure part once the test is failed.

Reply:
New code pushed, there is still something missing